src/noit_check_lmdb.c

```diff
@@ -0,0 +1,1571 @@
+/* Copyright (c) 2020, Circonus, Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ *       copyright notice, this list of conditions and the following
+ *       disclaimer in the documentation and/or other materials provided
+ *       with the distribution.
+ *     * Neither the name Circonus, Inc. nor the names of its contributors
+ *       may be used to endorse or promote products derived from this
+ *       software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "noit_check_lmdb.h"
+#include "noit_check.h"
+#include "noit_check_rest.h"
+#include "noit_lmdb_tools.h"
+#include <mtev_watchdog.h>
+
+#include <errno.h>
+
+static int noit_check_lmdb_show_check_json(mtev_http_rest_closure_t *restc, 
+                                           uuid_t checkid) {
+  return rest_show_check_json(restc, checkid);
+}
+
+static int noit_check_lmdb_add_attribute(xmlNodePtr root, xmlNodePtr attr, noit_lmdb_check_data_t *key_data, MDB_val mdb_data, bool separate_stanza) {
+  mtevAssert(root != NULL);
+  if ((!mdb_data.mv_data) || (mdb_data.mv_size == 0)) {
+    return 0;
+  }
+  if (!strcmp(key_data->key, "uuid")) {
+    /* This should be set separately */
+    return 0;
+  }
+  char *val = (char *)calloc(1, mdb_data.mv_size);
+  memcpy(val, mdb_data.mv_data, mdb_data.mv_size);
```

Comment:
is this null terminated, it doesn't seem that it would be... calloc(1 mdb_data.mv_size+1)? (and future occurences)

Reply:
Fixed